### PR TITLE
Adding heasarc tests 

### PR DESCRIPTION
**Pull request by nocturnalastro**
The local tests seem to be requiring remote data so I seem to have missed part of the local test setup but I can't see where.

### CODE REVIEW

File: astroquery/heasarc/tests/test_heasarc.py [hunk 1]
```diff
@@ -1,2 +1,82 @@
...
+
+def data_path(filename):
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    return os.path.join(data_dir, filename)
+
+
+@pytest.fixture
+def patch_get(request):
+    mp = request.getfuncargvalue("monkeypatch")
+    mp.setattr(requests, 'get', get_mockreturn)
```
Comment (bsipocz) [inline astroquery/heasarc/tests/test_heasarc.py]: this should patch ``Heasarc._request`` and not ``requests.get``, for examples see recent PRs #821 #822

File: astroquery/heasarc/tests/test_heasarc.py [hunk 2]
```diff
@@ -1,2 +1,82 @@
...
+
+
+# some basic defs of names
+good_object = "3c273"
+good_table = "rospublic"
+bad_object = "ImABadObject"
+bad_table = "ImABadTable"
+fallback_table = "numaster"
+
+def get_mockreturn(url, params=None, timeout=10, **kwargs):
```
Comment (bsipocz) [inline astroquery/heasarc/tests/test_heasarc.py]: The arguments should match that of ``Heasarc._request``

**Comment by bsipocz** [comment]
@nocturnalastro - This will probably work as expected when those two things are fixed.

File: astroquery/heasarc/core.py [hunk 3]
```diff
@@ -80,6 +84,14 @@ def _parse_result(self, response, verbose=False):
...
         except ValueError:
             return self._fallback(response.content)
+        except IORegistryError:
+            if _obj_not_found_msg in response.content:
+                object_name = response.url.split("Entry=")[-1].split("&")[0]
+                warnings.warn(AstropyWarning("Object {obj} ".format(obj=object_name) +
+                                             _obj_not_found_msg))
+            elif _table_not_found_msg in response.content:
+                table_name = response.url.split("BATCHRETRIEVALCATALOG_2.0+")[-1].split("&")[0]
+                warnings.warn(AstropyWarning(_table_not_found_msg + " for table {tbl}".format(tbl=table_name)))
```
Comment (keflavich) [inline astroquery/heasarc/core.py]: Should there be a catch-all 'else' here?

File: .eggs/README.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+This directory contains eggs that were downloaded by setuptools to build, test, and run plug-ins.
```
Comment (bsipocz) [inline .eggs/README.txt]: I'm not sure how this ended up being included (should have been filtered out via .gitignore), could you please remove the file from the commit?

**Comment by keflavich** [comment]
There are some minor changes that need to be made (e.g., untracking the `.eggs/readme.txt` file) before this can be merged, but otherwise is it ready?

**Comment by bsipocz** [comment]
This also needs a rebase to resolve the conflict.

**Comment by nocturnalastro** [comment]
I think I also need to fix a try/except I'll try to get it done this week :).

**Comment by bsipocz** [comment]
@nocturnalastro - Did you mean to close this after your last commit?

**Pull request closed without merging**